src/Router.js

```diff
@@ -1,36 +1,53 @@
 import React from 'react';
-import { Scene, Router, Stack } from 'react-native-router-flux';
+import { StyleSheet } from 'react-native';
+import { Scene, Router } from 'react-native-router-flux';
 import SignUpLogin from './components/auth/SignUpLogin';
 import LogIn from './components/auth/LogIn';
 import SignUp from './components/auth/SignUp';
 
+const styles = StyleSheet.create({
+  navigationBarStyle: {
+    backgroundColor: '#007fff',
+  },
+  titleStyle: {
+    color: 'white',
+  },
+});
+const commonSceneStyles = {
+  navigationBarStyle: styles.navigationBarStyle,
+  titleStyle: styles.titleStyle,
+  navBarButtonColor: 'white',
+};
+const scenesConfig = [
+  {
+    key: 'singUpLogin',
+    component: SignUpLogin,
+    initial: true,
+    onRight: () => {},
+    rightTitle: 'SKIP',
+    navTransparent: true,
+    rightButtonTextStyle: { color: 'white' },
+  },
+  {
+    key: 'signUp',
+    component: SignUp,
+    title: 'SIGN UP',
+    ...commonSceneStyles,
+  },
+  {
+    key: 'logIn',
+    component: LogIn,
+    title: 'LOG IN',
+    ...commonSceneStyles,
+  },
+];
 const RouterComponent = () => {
   return (
     <Router>
       <Scene>
-        <Scene
-          initial
-          onRight={() => {}}
-          rightTitle="SKIP"
-          key="signUpLogin"
-          component={SignUpLogin}
-          navTransparent="true"
-          rightButtonTextStyle={{ color: '#fff' }}
-        />
-        <Scene
-          key="LogIn"
-          component={LogIn}
-          title="LOG IN"
-          navigationBarStyle={{ backgroundColor: '#007fff' }}
-          titleStyle={{ color: 'white' }}
-        />
-        <Scene
-          key="SignUp"
-          component={SignUp}
-          title="SIGN UP"
-          navigationBarStyle={{ backgroundColor: '#007fff' }}
-          titleStyle={{ color: 'white' }}
-        />
+        {scenesConfig.map(scene => (
+          <Scene {...scene} />
```

Comment:
hmm... isn't `...commonSceneStyles` better off inside this `<Scene />`?
one time placement here vs adding it twice above (for now) inside `scenesConfig`

Reply:
ok, that does work. it would also apply to signUpLogin too, but i guess it is ok because i hid the nav

Reply:
dang it also, have a type in the signUpLogin key

Reply:
not going to implement suggestion, as signUpLogin doesn't use the common styles